lib/ret_web/controllers/api/v1/upload_controller.ex

```diff
@@ -0,0 +1,42 @@
+defmodule RetWeb.Api.V1.UploadController do
```

Comment:
hmm, I'm not sure if I would break this out into a new controller or not. the name "media controller" may be a bit of a misnomer, but in general I think the conceptual shoe fits to augment that controller to generate these database records.

Reply:
my expectation is there are going to be couple of cases here:

- ephemeral content uploads for in-room object spawning (current use case)
- permanent content uploads for in-room object spawning (persistent objects case, tbd semantics but uploads will land as permanent)
- conversion from ephemeral to permanent upload for in-room object spawning (a room or object converts from ephemeral to persistent)
- scene GLTF/GLB uploading, screenshot/profile image uploading

the last one is def a bit iffy, but the payload we return from the media API seems like it could work for all these cases.

Reply:
Marshall and I were actually leaning towards switching to the new /uploads/ API for media uploads as well. I think it makes sense to use a RESTful API for things that actually create records. Conceptually, it feels better to me than an amorphous /media/ api, but I could go either way.

Reply:
i see your point. i'll mull this over -- i can't quickly come up with a strong reason for either approach. one soft argument is that adding uploads to the existing API was really smooth, despite not being RESTful, so that might be at least part of an argument that its a good design.